news-aggregation/persistence/SubscriptionPersistence.ts

```diff
@@ -1,225 +1,213 @@
-            this.app
-                .getLogger()
-                .info("Could not delete subscription by id", err);
-        }
-    }
+	app: NewsAggregationApp;
+	persistenceRead: IPersistenceRead;
+	persistence: IPersistence;
+
+	constructor(
+		app: NewsAggregationApp,
+		persistenceRead: IPersistenceRead,
+		persistence: IPersistence
+	) {
+		this.app = app;
+		this.persistenceRead = persistenceRead;
+		this.persistence = persistence;
+	}
+
+	public async createSubscription(interval: string, user: IUser, room: IRoom) {
+		const associations: Array<RocketChatAssociationRecord> = [
+			new RocketChatAssociationRecord(
+				RocketChatAssociationModel.MISC,
+				'news-aggregation-subscription'
+			),
+			new RocketChatAssociationRecord(
+				RocketChatAssociationModel.MISC,
+				interval
+			),
+			new RocketChatAssociationRecord(RocketChatAssociationModel.USER, user.id),
+			new RocketChatAssociationRecord(RocketChatAssociationModel.ROOM, room.id),
+		];
+
+		let subscriptionRecord: ISubscription = {
+			userId: user.id,
+			interval: interval,
+			createdOn: new Date(),
+			user: user.id,
+			room: room.id,
+		};
+
+		let subscriptionId: string;
+		try {
+			subscriptionId = await this.persistence.createWithAssociations(
+				subscriptionRecord,
+				associations
+			);
+			console.log('subscription created!!', subscriptionId);
+		} catch (err) {
+			subscriptionId = '';
+			console.error('Could not create news subscription', err);
+			this.app.getLogger().info('Could not create news subscription', err);
+		}
+
+		return subscriptionId;
+	}
+
+	public async getSubscribedRooms(room: IRoom): Promise<Array<ISubscription>> {
+		const associations: Array<RocketChatAssociationRecord> = [
+			new RocketChatAssociationRecord(
+				RocketChatAssociationModel.MISC,
+				'news-aggregation-subscription'
+			),
+			new RocketChatAssociationRecord(RocketChatAssociationModel.ROOM, room.id),
+		];
+
+		let subscriptions: Array<ISubscription>;
+		try {
+			subscriptions = (await this.persistenceRead.readByAssociations(
+				associations
+			)) as Array<ISubscription>;
+		} catch (err) {
+			subscriptions = [];
+			console.error('Could not get subscribed rooms', err);
+			this.app.getLogger().info('Could not get subscribed rooms', err);
+		}
+
+		return subscriptions;
+	}
+
+	public async getSubscriptions(): Promise<Array<ISubscription>> {
+		const associations: Array<RocketChatAssociationRecord> = [
+			new RocketChatAssociationRecord(
+				RocketChatAssociationModel.MISC,
+				'news-aggregation-subscription'
+			),
+		];
+
+		let subscriptions: Array<ISubscription>;
+		try {
+			subscriptions = (await this.persistenceRead.readByAssociations(
+				associations
+			)) as Array<ISubscription>;
+		} catch (err) {
+			subscriptions = [];
+			console.error('Could not get subscriptions', err);
+			this.app.getLogger().info('Could not get subscriptions', err);
+		}
+
+		return subscriptions;
+	}
+
+	public async getSubscriptionById(
```

Comment:
There should be one `getSubscriptionsByInterval()` . This will help in subscription delivery service/job